fred2/missioncutscenesdlg.cpp

```diff
@@ -0,0 +1,466 @@
+
+	m_cutscenes_tree.SelectItem(h);
+}
+
+int CMissionCutscenesDlg::handler(int code, int node)
+{
+	switch (code) {
+		case ROOT_DELETED:
+			int i;
+			for (i = 0; i < (int)m_cutscenes.size(); i++){
+				if (m_cutscenes[i].formula == node) {
+					break;
+				}
+			}
+
+			Assert(i < (int)m_cutscenes.size());
+			m_cutscenes.erase(m_cutscenes.begin() + i);
+			m_sig.erase(m_sig.begin() + i);
+
+			return node;
+
+		default:
+			Int3();
+	}
+
+	return -1;
+}
+
+void CMissionCutscenesDlg::OnSelchangeCutsceneTypeDrop()
+{
+	if (cur_cutscene < 0) {
+		return;
+	}
+
+	UpdateData(TRUE);
+	UpdateData(TRUE);  // doesn't seem to update unless we do it twice..
+
+	m_cutscenes[cur_cutscene].type = m_cutscene_type;
+
+	HTREEITEM h = m_cutscenes_tree.GetSelectedItem();
+	Assert(h);
+
+	HTREEITEM h2;
+	while ((h2 = m_cutscenes_tree.GetParentItem(h)) != 0) {
+		h = h2;
+	}
+
+	m_cutscenes_tree.DeleteItem(h);
+	cur_cutscene = -1;
+	update_cur_cutscene();
+}
+
+void CMissionCutscenesDlg::OnChangeCutsceneName()
+{
+	if (cur_cutscene < 0) {
+		return;
+	}
+
+	UpdateData(TRUE);
+	HTREEITEM h = m_cutscenes_tree.GetSelectedItem();
+	if (!h){
+		return;
+	}
+
+	HTREEITEM h2;
+	while ((h2 = m_cutscenes_tree.GetParentItem(h)) != 0) {
+		h = h2;
+	}
+
+	m_cutscenes_tree.SetItemText(h, m_name);
+	strcpy_s(m_cutscenes[cur_cutscene].filename, m_name);
+}
+
+void CMissionCutscenesDlg::OnCancel()
+{
+	theApp.record_window_data(&Messages_wnd_data, this);
+	CDialog::OnCancel();
+}
+
+void CMissionCutscenesDlg::OnClose()
+{
+	if (query_modified()) {
+		int z = MessageBox("Do you want to keep your changes?", "Close", MB_ICONQUESTION | MB_YESNOCANCEL);
+		if (z == IDCANCEL)
+			return;
+
+		if (z == IDYES) {
+			OnOk();
+			return;
+		}
+	}
+	
+	CDialog::OnClose();
+}
+
+void CMissionCutscenesDlg::insert_handler(int old, int node)
+{
+	int i;
+
+	for (i = 0; i < (int)m_cutscenes.size(); i++) {
+		if (m_cutscenes[i].formula == old) {
+			break;
+		}
+	}
+
+	Assert(i < (int)m_cutscenes.size());
+	m_cutscenes[i].formula = node;
+	return;
+}
+
+void CMissionCutscenesDlg::move_handler(int node1, int node2, bool insert_before)
+{
+	int index1;
+	for (index1 = 0; index1 < (int)m_cutscenes.size(); index1++) {
+		if (m_cutscenes[index1].formula == node1) {
+			break;
+		}
+	}
+	Assert(index1 < (int)m_cutscenes.size());
+
+	int index2;
+	for (index2 = 0; index2 < (int)m_cutscenes.size(); index2++) {
+		if (m_cutscenes[index2].formula == node2) {
+			break;
+		}
+	}
+	Assert(index2 < (int)m_cutscenes.size());
+
+	mission_cutscene fu = m_cutscenes[index1];
+	int s = m_sig[index1];
+
+	int offset = insert_before ? -1 : 0;
+
+	while (index1 < index2 + offset) {
+		m_cutscenes[index1] = m_cutscenes[index1 + 1];
+		m_sig[index1] = m_sig[index1 + 1];
+		index1++;
+	}
+	while (index1 > index2 + offset + 1) {
+		m_cutscenes[index1] = m_cutscenes[index1 - 1];
+		m_sig[index1] = m_sig[index1 - 1];
+		index1--;
+	}
+
+	m_cutscenes[index1] = fu;
+	m_sig[index1] = s;
+}
```

Comment:
There are a couple more casts and asserts in this function.  Sorry ....

Reply:
No problem. I think I got them all. Converted most to size_t, used a few static_casts where needed. Updated asserts to assertions, etc. Let me know if there's anything I missed.